connector_onpay/wizards/onpay_data_export.py

```diff
@@ -0,0 +1,236 @@
+# Copyright (C) 2020 Open Source Integrators
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+import base64
+import csv
+import os
+import tempfile
+
+from odoo import api, fields, models
+from odoo.tools import misc
+
+import xlsxwriter
+
+
+class OnPayExport(models.TransientModel):
+    _name = 'onpay.export'
+    _description = "OnPay Export Wizard"
+
+    date_from = fields.Date(
+        string='From'
+    )
+    date_to = fields.Date(
+        string='To'
+    )
+    file_type = fields.Selection(
+        [('csv', 'CSV'),
+         ('xls', 'XLS')], default='csv', string="File Type")
+
+    @api.multi
+    def export_data(self):
+        if self.file_type == 'csv':
+            res_id, context = self.export_data_csv()
+        else:
+            res_id, context = self.export_data_xls()
+        return {
+            'type': 'ir.actions.act_window',
+            'view_type': 'form',
+            'view_mode': 'form',
+            'res_model': 'wiz.xls.content',
+            'target': 'new',
+            'res_id': res_id.id,
+            'context': context,
+        }
+
+    @api.multi
+    def export_data_csv(self):
+        if self._context.get('active_ids'):
+            emp_obj = self.env['hr.employee']
+            onpay_type_obj = self.env['onpay.pay.type']
+
+            # Timesheet data of employees between selected date
+            self._cr.execute("select employee_id,line_onpay_id,\
+                sum(unit_amount) as unit_amount from \
+                account_analytic_line where \
+                date >= %s and date <= %s and employee_id in %s \
+                group by employee_id,line_onpay_id", (
+                self.date_from, self.date_to,
+                tuple(self._context.get('active_ids')),))
+            my_data = self._cr.dictfetchall()
+
+            # Expense data (approved) of employees between selected date
+            self._cr.execute("select employee_id,expense_onpay_id,\
+                sum(total_amount) as \
+                total_amount from \
+                hr_expense where date >= %s and \
+                date <= %s and state='approved' and employee_id in %s \
+                group by employee_id,expense_onpay_id", (
+                self.date_from, self.date_to,
+                tuple(self._context.get('active_ids')),))
```

Comment:
I wonder if these don't also post analytic lines, and we end up with duplicate data?